### PR TITLE
Survey controller

### PR DESCRIPTION
**Pull request by Brunation11**

### CODE REVIEW

File: app/controllers/surveys_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# find chosen survey and redirect to its questions
+get 'survey/:id' do
+  survey = TakenSurvey.new(survey_id: params[:id], user_id: current_user.id)
```
Comment (aquarian5) [inline app/controllers/surveys_controller.rb:3]: user_id: current_user

Comment (Brunation11) [inline app/controllers/surveys_controller.rb:3]: thank you for the review, according to the helper method, current_user is just set to User.find so it does need id to be called on it

File: app/controllers/questions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# show each individual question
+get '/questions/:id' do
+  question = Question.find(params[:id] + 1)
```
Comment (aquarian5) [inline app/controllers/questions_controller.rb]: don't know why we are incrementing in the controller :dancer: :dancers:

File: app/controllers/questions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  survey = TakenSurvey.find(params[:survey_id])
+  if defined?(question)
+    erb :'/questions/show', locals: {question: question, survey: survey}
+  else
+    erb :'/stats'
```
Comment (aquarian5) [inline app/controllers/questions_controller.rb]: sweet idea, I think '/stats' should be '/answers'

File: app/controllers/questions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  end
+end
+
+# post to the answers table to track a users response for a specific question then redirect to the next question, else return to the same page and
+post '/questions' do
+  answer = Answer.new(params)
+  if answer.save
+    redirect '/questions/<%= params[:question.id] %> + 1'
```
Comment (aquarian5) [inline app/controllers/questions_controller.rb]: you are incrementing here (so incrementing twice) incrementing should be in the model if you are incrementing, as a method  :dash:  :lips:

**Pull request closed without merging**